mypy/plugins/ctypes.py

```diff
@@ -0,0 +1,186 @@
+"""Plugin to provide accurate types for some parts of the ctypes module."""
+
+from typing import Optional
+
+# Fully qualified instead of "from mypy.plugin import ..." to avoid circular import problems.
+import mypy.plugin
+from mypy.subtypes import is_subtype
+from mypy.types import CallableType, Instance, Type, UnionType
+
+
+def _find_simplecdata_base_arg(tp: Instance) -> Optional[Type]:
+    """Try to find a parametrized _SimpleCData in tp's bases and return its single type argument.
+
+    None is returned if _SimpleCData appears nowhere in tp's (direct or indirect) bases,
+    or if it doesn't have a single type argument.
+    """
+
+    for mro_type in tp.type.mro:
+        for base in mro_type.bases:
+            if len(base.args) == 1:
+                return base.args[0]
```

Comment:
> This will find any generic type with a single argument, aren't we looking only for `_SimpleCData`?

You're completely right, I forgot to check the type name.

> `mypy.maptype.map_instance_to_supertype`

Exactly what I needed here, thanks for the tip!